src/domain/navigation/index.ts

```diff
@@ -217,15 +222,28 @@ export function stringifyPath(path: Path<SegmentType>): string {
                 continue;
             default:
                 urlPath += `/${segment.type}`;
-                if (segment.value && segment.value !== true) {
-                    urlPath += `/${segment.value}`;
+                if (encodedSegmentValue && encodedSegmentValue !== true) {
+                    urlPath += `/${encodedSegmentValue}`;
                 }
         }
         prevSegment = segment;
     }
     return urlPath;
 }
 
+function encodeSegmentValue(value: SegmentType[keyof SegmentType]) {
+    if (typeof value === "boolean") {
+        // Nothing to encode for boolean
+        return value;
+    }
+    else if (Array.isArray(value)) {
+        return value.map(v => encodeURIComponent(v));
```

Comment:
could do the `.join(",")` straight here, as it's part of the encoding it to a string process...